### PR TITLE
Make the reqwest client static and reuse it for all requests

### PR DESCRIPTION
**Pull request by coltfred**
Currently I'm caching this globally, which allows us to share it for user create as well as for all Ironoxide instances. This has the following advantages:
- Shared even on user create, etc
- Shared even if the user creates new SDKs.

### CODE REVIEW

**Comment by coltfred** [comment]
Note that I believe that `.clone` on the Reqwest Client just increments the Arc counter since the Client just holds an Arc. https://docs.rs/reqwest/0.10.0-alpha.2/src/reqwest/async_impl/client.rs.html#49

File: src/internal/mod.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ use std::{
 
 pub mod document_api;
 pub mod group_api;
-mod rest;
+pub(crate) mod rest; //COLT: Feels a bit bad.
```
Comment (clintfred) [inline src/internal/mod.rs]: an alternative would be to allow the internal API to generate the `IronCoreRequest` struct, allowing it to stay private to internal.

File: src/internal/mod.rs [hunk 2]
```diff
@@ -38,7 +38,8 @@ lazy_static! {
         .to_string(),
         _ => "https://api.ironcorelabs.com/api/1/".to_string(),
     };
-    pub static ref OUR_REQUEST: IronCoreRequest = IronCoreRequest::new(URL_STRING.as_str());
+    // pub static ref OUR_REQUEST: IronCoreRequest =
```
Comment (clintfred) [inline src/internal/mod.rs]: I wonder if it would be appropriate to open an issue on reqwest to better understand why this doesn't work?

**Comment by coltfred** [comment]
At this point we don't have a great understanding about the contention issues in highly concurrent situations, so we're going to just close this. The reason we were trying it in the first place was because we wanted performance gains, but without that as a driving factor it doesn't seem like it's worth putting time into right now.

**Pull request closed without merging**